acceptance/analyzer_test.go

```diff
@@ -395,63 +414,117 @@ func testAnalyzerFunc(platformAPI string) func(t *testing.T, when spec.G, it spe
+					fmt.Sprintf("%s %s %s; ls -al /some-dir",
 						ctrPath(analyzerPath),
-						imageName,
+						flatPrint(analyzeArgs),
+						authRegistry.RepoName("some-image"),
 					),
 				),
 			)
 
 			h.AssertMatch(t, output, "2222 3333 .+ some-analyzed.toml")
 		})
 
-		// TODO: add with or after https://github.com/buildpacks/lifecycle/pull/646
-		when.Pend("run image", func() {
+		when("run image", func() {
 			when("provided", func() {
-				it("validates read access", func() {
+				it("is recorded in analyzed.toml", func() {
 					h.SkipIf(t, api.MustParse(platformAPI).Compare(api.MustParse("0.7")) < 0, "Platform API < 0.7 does not accept run image")
+
+					h.DockerRunAndCopy(t,
+						containerName,
+						copyDir,
+						ctrPath("/layers/analyzed.toml"),
+						analyzeImage,
+						h.WithFlags(
+							"--env", "CNB_PLATFORM_API="+platformAPI,
+							"--env", "CNB_REGISTRY_AUTH="+fixtures.authRegAuthConfig,
+							"--network", registryNetwork,
+						),
+						h.WithArgs(ctrPath(analyzerPath), "-run-image", fixtures.authRegRunImage, authRegistry.RepoName("some-image")),
+					)
+
+					analyzedMD := assertAnalyzedMetadata(t, filepath.Join(copyDir, "analyzed.toml"))
+					h.AssertEq(t, analyzedMD.RunImage.Reference, fixtures.authRegRunImage)
 				})
 			})
 
 			when("not provided", func() {
 				it("falls back to CNB_RUN_IMAGE", func() {
 					h.SkipIf(t, api.MustParse(platformAPI).Compare(api.MustParse("0.7")) < 0, "Platform API < 0.7 does not accept run image")
+
+					h.DockerRunAndCopy(t,
+						containerName,
+						copyDir,
+						ctrPath("/layers/analyzed.toml"),
+						analyzeImage,
+						h.WithFlags(
+							"--env", "CNB_PLATFORM_API="+platformAPI,
+							"--env", "CNB_REGISTRY_AUTH="+fixtures.authRegAuthConfig,
+							"--env", "CNB_RUN_IMAGE="+fixtures.authRegRunImage,
+							"--network", registryNetwork,
+						),
+						h.WithArgs(ctrPath(analyzerPath), authRegistry.RepoName("some-image")),
+					)
+
+					analyzedMD := assertAnalyzedMetadata(t, filepath.Join(copyDir, "analyzed.toml"))
+					h.AssertEq(t, analyzedMD.RunImage.Reference, fixtures.authRegRunImage)
 				})
 
 				when("CNB_RUN_IMAGE not provided", func() {
 					it("falls back to stack.toml", func() {
 						h.SkipIf(t, api.MustParse(platformAPI).Compare(api.MustParse("0.7")) < 0, "Platform API < 0.7 does not accept run image")
+
+						cmd := exec.Command("docker", "run", "--rm",
+							"--env", "CNB_PLATFORM_API="+platformAPI,
+							"--env", "CNB_REGISTRY_AUTH="+fixtures.authRegAuthConfig,
+							"--network", registryNetwork,
+							analyzeImage,
+							ctrPath(analyzerPath),
+							"-stack", "/cnb/platform-0.7-stack.toml",
+							authRegistry.RepoName("some-image"),
+						) // #nosec G204
+						output, err := cmd.CombinedOutput()
+						h.AssertNotNil(t, err)
+
+						h.AssertStringContains(t, string(output), "failed to : ensure registry read access to some-run-image")
```

Comment:
I'm confused. Can you please explain why we expect an error in this case?

Reply:
I added a comment on line 497. Does that help?

Reply:
Updated in 07e41be

Reply:
Are we supposed to get a failure in this test case? According to the test name, we're supposed to "falls back to stack.toml" and not "error", no?

Reply:
It falls back to stack.toml, but it fails because the run image that stack.toml references doesn't exist.

Reply:
I'll add a TODO to update this test once we have @jjbustamante's test helper improvements. That will make it clearer.

Reply:
Please see 20b0608

Reply:
I understand that it fails because the run image doesn't exist. My point was that it's confusing that we see a failure in a test that is supposed to pass.
But I'm totally fine with waiting for Juan's test helper improvements.
Thanks!